plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -502,6 +504,40 @@ public Optional<ConnectorOutputMetadata> finishInsert(ConnectorSession session,
         return Optional.empty();
     }
 
+    @Override
+    public void setTableComment(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<String> newComment)
+    {
+        BigQueryTableHandle table = (BigQueryTableHandle) tableHandle;
+        BigQueryClient client = bigQueryClientFactory.createBigQueryClient(session);
+
+        RemoteTableName remoteTableName = table.asPlainTable().getRemoteTableName();
+        String sql = format(
+                "ALTER TABLE %s.%s.%s SET OPTIONS (description = %s)",
+                quote(remoteTableName.getProjectId()),
+                quote(remoteTableName.getDatasetName()),
+                quote(remoteTableName.getTableName()),
+                newComment.map(comment -> "'" + comment + "'").orElse("NULL"));
```

Comment:
what if comment contains ' ?

Reply:
Good catch. I changed to parameterized query and confirmed manually.
```sql
COMMENT ON TABLE test IS 'an''apostrophe';
COMMENT ON COLUMN test.x IS 'an''apostrophe'

SHOW CREATE TABLE test;
              Create Table
-----------------------------------------
 CREATE TABLE bigquery.ebyhr_test.test (
    x bigint COMMENT 'an''apostrophe'
 )
 COMMENT 'an''apostrophe'
(1 row)
```

Filed #14058 for follow-up test cases.